Update password reset functionality

Hordunlarmy opened the pull request:
Update password reset functionality to use email for change password validation

JoseMiracle reviewed (commented):
I think I have worked on this, check the PR I made some hours ago  regarding refactoring authentication app.

Hordunlarmy commented:
> I think I have worked on this, check the PR I made some hours ago regarding refactoring authentication app.

made little additional changes.

- changed the method from post to put
- replaced old_password field with email in serializer
- updated reset url too from password/change to password/reset
- Updated tests the match the changes
